seps/sep-0002.md

```diff
@@ -0,0 +1,169 @@
+1. Leaving a group will stop content from appearing on a user’s feed.
+1. Group content should only appear in the “All” activity feed if a group member reposts a group post to the main feed.
+1. One should be able to retrieve a list of all groups in socialDB
+1. One should be able to retrieve a list of all users in a group.
+1. One should be able to search for groups, where search text is matched against the group's name and metadata.
+
+
+## Rationale and Alternatives
+
+A key success metric of this standard is to provide a minimal blueprint that gateways can build upon to ensure feature parity, in the UX goals above, across gateways. 
+
+Prior art: 
+
+1. near.social's NFT-based groups
+
+The [group's post page](https://near.org/near/widget/ComponentDetailsPage?src=mob.near/widget/N.Group&tab=source); passing to this page an NFT's contractID as the groupId query param will [render the group details](https://near.org/mob.near/widget/N.Group?groupId=v0.8bityonce.near).   A primary drawback to this NFT-based approach is that there is no canonical list of group members. Instead, group membership is inferred from the owners of a NFT per a non-socialDB API - a Mintbase API is used in the above component. 
+
+
+1. devs.near's groups [defined implicitly by follow-style graph connections](https://near.org/devs.near/widget/every.group). 
+
+A primary drawback to this graph-based approach is that there is no canonical source of group membership. A single group of N members can have M distinct versions of who belongs in that group, where M is the number of group members who decline to "mutually follow" each newly requested group member.  Group membership is managed by an object on the namespace of anyone who has requested access to a group, which contains a group identifier as a key with a value of a list of accountIds (AccountList). Even if no one approves someone's request to join a group, the requestor will retain a valid instance of that groupIdentifer:AccountList on their socialDB namespace - with their accountID included as a group member. This results in muliple versions of graphs which represent the definition of members of a group. For more details on this implementation, see minute 33:45 [here](https://youtu.be/HAyweTF7IjU?si=z_hfct3oK_86nt_o)
+
+With this approach, can we efficiently determine which group members should receive a notification when a post is created on that group's page?  How challenging would it be to ensure that only group members can post to this group's page? There is a solution here, however these implicitly defined group connections with multiple versions of group membership adds complexity. 
+
+
+1. [Communities on DevHub](https://near.org/devgovgigs.near/widget/Ideas)
+
+DevHub's community implementation exists on a distinct contract from near.social. This results in a lack of discoverability of DevHub's groups across gateways due to its community data not being included in the indexes which power search on socialdB.
```

Comment:
One workaround of this problem is implement group on contract side and define a common interface that custom indexer should implementation, a minimum example could be `create_group`, `join_group` and `leave_group` calls should be indexed. And the gateway can then rely on the indexed data to form discovery. The permission logic is still handled by custom contract